posit-dev/positron
# Feature/dynamic plot zoom

Address #4210 

Reuses the component created for static plots to enable pan and zoom. Modified the component to recalculate the image scaling so that the image is properly displayed when switching between plots.

Fixed a warning about multi-window in the dynamic plot component. It might have been a problem in the future with multi-window instances. Also, fixed a refresh problem with the component re-rendering unnecessarily due to missing hook dependencies.

![image](https://github.com/user-attachments/assets/ccdbc2ea-0f7b-4e4e-86dc-af9d2c34236f)

```R
library(datasets)
hist(iris$Sepal.Width)
```
### QA Notes
An area to focus on is changing the plot. The zoom and sizing policy should update after changing plots if they were changed while viewing the previous plot.

Smoke tests were updated to find the plots since the structure changed to encapsulate the image in a scrollable element. Also changed the check on the zoom option to ensure it is visible with a dynamic plot.
